mansur_adam_ndvi.py

```diff
@@ -0,0 +1,892 @@
+
+
+# This cell  is testing your data output above
+
+student_ndvi_ts_single_site = _
+
+single_scene_points = 0
+
+# Ensure the data is stored in a dataframe.
+if isinstance(student_ndvi_ts_single_site, pd.DataFrame):
+    print('\u2705 Your data is stored in a DataFrame!')
+    single_scene_points += 1
+else:
+    print('\u274C It appears your data is not stored in a DataFrame. ',
+          'To see what type of object your data is stored in, check its type with type(object)')
+
+# Ensure that the date column is the index
+if isinstance(student_ndvi_ts_single_site.index, pd.core.indexes.datetimes.DatetimeIndex):
+    print('\u2705 You have the index set to the date column!')
+    single_scene_points += 2
+else:
+    print('\u274C You do not have the index set to the date column.')
+
+# Ensure that the date column is datetime
+if isinstance(student_ndvi_ts_single_site.index[0], pd._libs.tslibs.timestamps.Timestamp):
+    print('\u2705 The data in your date column is datetime!')
+    single_scene_points += 2
+else:
+    print('\u274C The data in your date column is not datetime.')
+
+# Ensure the site name is correct
+if student_ndvi_ts_single_site.site.values[0] == 'HARV':
+    print('\u2705 You have the correct site name!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct site name.')
+
+if np.allclose(0.281131628228094, student_ndvi_ts_single_site.mean_ndvi.values[0]):
+    print('\u2705 You have the correct mean NDVI value!')
+    single_scene_points += 5
+else:
+    print('\u274C You do not have the correct mean ndvi value.')
+
+print("\n \u27A1 You received {} out of 15 points for creating a dataframe.".format(
+    single_scene_points))
+single_scene_points
+
+
+# ## Task 2:
+# 
+# In the cell below, process all of the landsat scenes. Create a DataFrame that contains the following 
+# information for each scene
+# 
+# 
+# |   | index  | site  | mean_ndvi  | 
+# |---|---|---|---|
+# | Date  |   |   |   |
+# |  2017-01-07  | 0  | SJER  | .4  |  
+# 
+# Be sure to call your dataframe at the end of the cell to ensure autograding works.
+# HINT: FOR THIS STEP, leave any rows containing missing values (`NAN`).
+
+# ### Data sources
+# 
+# Data for the following calculation of NDVI was collected by the Landsat 8
+# satellite in 2017 for the HARV and SJER field sites as part of the NEON program.
+
+# In[8]:
+
+
+# Create dataframe of NDVI including the cleaning data to deal with clouds
+
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Don't forget to set date as the index and make the values of type datetime
+
+# Define path to SJER folder (the path to HARV is defined above)
+sjer_path = os.path.join("ndvi-automation", "sites", "SJER")
+
+# Read boundaries for each site
+boundaries = {
+    "HARV": gpd.read_file(os.path.join(harv_path, "vector", "HARV-crop.shp")),
+    "SJER": gpd.read_file(os.path.join(sjer_path, "vector", "SJER-crop.shp"))
+}
```

Comment:
I like how you put the boundary paths into a dictionary, but what if you added a site? You would have to manually add it to this dictionary. I think this can be made more generic so the mapping to the each boundary shapefile could apply to any number of sites (assuming file structures and naming conventions are the same, of course).